### ISSUE TITLE
Cannot fully build with cmake

### ISSUE THREAD

**Issue by charlie090** [ethereum-mining/ethminer#2163]
**Describe the bug**
Cannot build ethminer with configurations

**To Reproduce**
I am at the stage of building ethminer in the newly created [build] directory: cmake ..

I get an error message:
CMake Error at /usr/share/cmake-3.16/Modules/FindCUDA.cmake:707 (message):
  Specify CUDA_TOOLKIT_ROOT_DIR
Call Stack (most recent call first):
  libhwmon/CMakeLists.txt:19 (find_package)

To be sure I have tried installing the nvidia CUDA toolkit but could not as it is not compatible in my specific GPU(?).

So now I am trying to build without CUDA components.

I have edited the "CMakeLists.txt" file in ethminer directory to say: option(ETHASHCUDA "Build with CUDA mining" OFF)

However, this line edit seems to not have any material change as it not only keeps the same error message but also does not change the build configurations.

Currently, I am a bit lost and might have to reinstall Ubuntu.

**Expected behavior**
I should be able to successfully build

**Screenshots (Optional)**

**Environment (please complete the following information):**
- Operating System: Ubuntu 20.04.2 LTS (Kernel: 5.4.0-54-generic)
- Hardware AMD Radeon VII
- Ethminer Version 0.18
- Ethminer options used

**Additional context**
Default release build configurations:
option(ETHASHCL "Build with OpenCL mining" ON)
option(ETHASHCUDA "Build with CUDA mining" ON)
option(ETHASHCPU "Build with CPU mining (only for development)" OFF)
option(ETHDBUS "Build with D-Bus support" OFF)
option(APICORE "Build with API Server support" ON)
option(BINKERN "Install AMD binary kernels" ON)
option(DEVBUILD "Log developer metrics" OFF)
option(USE_SYS_OPENCL "Build with system OpenCL" OFF)

Current build configurations:
option(ETHASHCL "Build with OpenCL mining" ON)
option(ETHASHCUDA "Build with CUDA mining" OFF) ##### <----changed here
option(ETHASHCPU "Build with CPU mining (only for development)" OFF)
option(ETHDBUS "Build with D-Bus support" OFF)
option(APICORE "Build with API Server support" ON)
option(BINKERN "Install AMD binary kernels" ON)
option(DEVBUILD "Log developer metrics" OFF)
option(USE_SYS_OPENCL "Build with system OpenCL" OFF)

**Comment by charlie090** [comment]
I have been able to build after installing the nvidia cuda toolkit but I think this issue is still valid.

**Comment by ewook** [comment]
I agree with the creator of the issue. The intricate web of dependencies creates a scenario where build-options are overruled.
I am trying to build only for OpenCL and end up in the same situation.

**Comment by ewook** [comment]
I've found a solution - disregard the "source" packages in the repository. Clone the whole etherminer. Run the git portion described in the README, and it should compile. This should indicate that the source packages are broken.

**Comment by Grimmjow-J** [comment]
hello, i only get 46mh/s with ethminer on ubuntu 20.04, do you have this problem?

**Comment by Grimmjow-J** [comment]
> hello, i only get 46mh/s with ethminer on ubuntu 20.04, do you have this problem?
my card is radeon vii
